pkg/scm/git/git.go

```diff
@@ -48,6 +48,8 @@ type URLMods struct {
 	Path   string
 	// Corresponds to Fragment in the URL struct, but we use Ref since URL Fragments are used and git commit refs
 	Ref string
+	// for use by scm.go DownloaderForSource for updating source input parameter into the s2i command
+	NewSource string
```

Comment:
since this appears to only be set/used when the original path was a local filesystem path, i think the name needs work.  (as does the comment).

alternatively change the instantiation of URLMods such that this is always populated.